### PR TITLE
Tweaks to messenger docs based on feedback from @Tobion

### PR DESCRIPTION
**Pull request by weaverryan**
See @Tobion's review: https://github.com/symfony/symfony-docs/pull/11602#pullrequestreview-244097540 - it contains some good stuff.

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

File: messenger.rst [hunk 1]
```diff
@@ -1067,7 +1076,7 @@ A handler class can handle multiple messages or configure itself by implementing
 
 .. _messenger-handlers-different-transports:
 
-Sending Handlers to Different Transports
+Binding Handlers to Different Transports
```
Comment (Tobion) [inline messenger.rst]: there is a ref. that also says `sending handlers to different transports`

File: messenger.rst [hunk 2]
```diff
@@ -1243,8 +1252,9 @@ for each bus looks like this:
 
 #. ``dispatch_after_current_bus``- see :doc:`/messenger/message-recorder`.
 
-#. ``failed_message_processing_middleware`` - sends failed messages to the
-   :ref:`failure transport <messenger-failure-transport>`.
+#. ``failed_message_processing_middleware`` - processes messages that are being
+   retried via the :ref:`failure transport <messenger-failure-transport>` to make
+   them properly function as if they were being received from their original transport.
 
 #. Your own collection of middleware_;
```
Comment (wouterj) [inline messenger.rst]: Just spotted the inconsistent usage of `;` or `.` in this list. According to our "very strict" guideline on this, every list items should end with a semicolon except from the last one.

**Review by Tobion — APPROVED** [review]

File: messenger.rst [hunk 3]
```diff
@@ -709,7 +709,11 @@ to retry them:
     $ php bin/console messenger:failed:retry 20 30 --force
 
     # remove a message without retrying it
-    $ php bin/console messenger:failed:retry 20
+    $ php bin/console messenger:failed:remove 20
+
+If the messages fails again, it will be re-sent back to the failure transport
```
Comment (javiereguiluz) [inline messenger.rst]: `messages fails` -> `messages fail`

**Review by javiereguiluz — APPROVED** [review]
Thanks Ryan and Tobias.

Note to mergers: please, fix a minor typo while merging. Thanks

**Pull request merged**